src/v/raft/replicate_entries_stm.cc

```diff
@@ -260,13 +267,15 @@ inline bool replicate_entries_stm::should_skip_follower_request(vnode id) {
               id);
             return true;
         }
-        if (f_meta.last_sent_offset != _meta.prev_log_index) {
+        if (
+          f_meta.in_flight_offset
+          && f_meta.in_flight_offset != _meta.prev_log_index) {
```

Comment:
I think when in_flight_offset is null, we should check next_index as well. Otherwise we'll always be sending replicate requests between recovery requests, right?

Reply:
we can not check `next`, as next is updated after append entries reply is received by leader but here we may have more than one request pending for the follower